cw/cw305/cw_lite_segmented.py

```diff
@@ -0,0 +1,175 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Support for capturing traces using ChipWhisperer-Lite in segmented mode."""
+
+import logging
+import re
+import time
+
+import chipwhisperer as cw
+import numpy as np
+from numpy.lib.stride_tricks import as_strided
+
+
+class CwLiteSegmented:
+    """Class for capturing traces using a ChipWhisperer-Lite.
+
+    This class uses segmented traces mode to improve capture performance.
+
+    When in segmented mode, ChipWhisperer-Lite captures multiple waves each starting at
+    a trigger event. This is much more efficient than sending a separate command for
+    each wave.
+
+    Due to current limitations of the hardware, number of waves (segments) per capture
+    is fixed and there is also a lower limit on the number of samples per wave. This
+    class handles this internally but also makes this information available via the
+    read-only ``num_segments_actual`` attribute so that the target can be configured to
+    generate the correct number of triggers.
```

Comment:
I would clarify this section e.g. as follows:

"Due to current limitations of the firmware, ChipWhisperer-Lite always needs to fill the entire sample buffer when running in batch mode. Depending on the number of samples per segment, fewer or more segments need to be captured to fill the sample buffer, and the last segment may be captured only partially. Internally, this class handles all the associated processing including discarding partially captured segments.

However, it must be ensured that the target indeed generates the correct number of capture triggers to fill the sample buffer. This number is made available via the read-only ``num_segments_actual`` attribute so that the target can be configured accordingly."

Reply:
Done, thanks.